packages/devtools_app/lib/src/extensions/embedded/_view_web.dart

```diff
@@ -156,14 +156,14 @@ class _ExtensionIFrameController extends DisposableController
       ' _iFrameReady future completed.',
     );
     embeddedExtensionController.extensionIFrame.contentWindow!.postMessage(
-      message,
-      embeddedExtensionController.extensionUrl,
+      message.jsify(),
+      embeddedExtensionController.extensionUrl.toJS,
     );
   }
 
-  void _handleMessage(html.Event e) {
-    if (e is html.MessageEvent) {
-      final extensionEvent = DevToolsExtensionEvent.tryParse(e.data);
+  void _handleMessage(Event e) {
+    if (e is MessageEvent) {
```

Comment:
This is one of the big pitfalls of interop alas. For context: in `dart:html`, we did the expensive checking to ensure that a `MessageEvent` is *actually* a `MessageEvent` under the hood. With `package:web`, these are all just different views of a JS object, so this `is` check will always pass, even if `e` is not a `MessageEvent`. This will become a bit more obvious once we generate `package:web` types using extension types with a representation type of `JSObject`. Hopefully there should be a lint here telling you that you're actually doing a `is JSObject` check, reducing the confusion even further.

The right way to do this is `(e as JSObject).instanceOfString('MessageEvent')` instead. When `Event` is an extension type on `JSObject`, you won't need the cast.

Reply:
Since `Event` implements `JSObject`, do I still need the cast? I'm getting an unnecessary cast lint when I add it.

Reply:
Ah no, I forgot that we subtype it. You can just do `e.instanceOfString('MessageEvent')`.